chemicaljson.py

```diff
@@ -0,0 +1,256 @@
+
+
+class Coords(BaseModel):
+    """Coordinates for the atoms in the molecule.
+
+    Length must match the number of atoms*3 (x, y, z).
+    """
+
+    field_3d: List[float] = Field(..., alias="3d", description="List of 3d Cartesian coordinates (in Angstrom) for the atoms [ x, y, z, x, y, z, ... ]")
+    field_3dFractional: Optional[List[float]] = Field(None, alias="3dFractional", description="Optional list of 3d fractional coordinates for the atoms [ x, y, z, x, y, z, ... ]")
+    field_3dSets: Optional[List[List[float]]] = Field(None, alias="3dSets", description="Optional list of lists of 3d Cartesian coordinates (in Angstrom) for the atoms [ [x, y, z], [x, y, z], ... ]")
+
+
+class Atoms(BaseModel):
+    """Atoms in the molecule."""
+
+    elements: Elements = Field(..., description="List of atomic numbers for the atoms.")
+    coords: Coords = Field(..., description="List of coordinates.")
+    formalCharges: Optional[List[int]] = Field(None, description="Optional list of formal charges for the atoms.")
+    labels: Optional[List[str]] = Field(None, description="Optional list of custom labels for atoms (e.g., 'R' / 'S' or '0.12', etc.)")
+    layer: Optional[List[int]] = Field(None, description="Optional list of layer numbers for the atoms (generally just 1 for most molecules).")
+
+
+class Connections(BaseModel):
+    """Connections - list of connections between atom indices.
+
+    Length must be the number of bonds * 2
+    """
+
+    index: List[int]
+
+
+class Bonds(BaseModel):
+    """
+    Optional bonds between atoms, including connections and bond orders for the atoms in the molecule. (Optional)
+    """
+
+    connections: Connections
+    order: List[int]
+
+
+class BasisSet(BaseModel):
+    """Basis Set information (optional)
+
+    At the moment, implied to be Gaussian basis sets.
+    """
+
+    coefficients: List[float] = Field(..., description="List of coefficients for the basis functions.")
+    exponents: List[float] = Field(..., description="List of exponents for the basis functions.")
+    primitivesPerShell: List[int] = Field(..., description="List of number of primitives per shell.")
+    shellToAtomMap: List[int] = Field(..., description="List of atom indices for the basis functions.")
+    shellTypes: List[int] = Field(..., description="List of shell types for the basis functions (l-value, so s=0, p=1, d=2, etc.).")
+
+
+class Orbitals(BaseModel):
+    """Information about molecular orbital energies and coefficients. (Optional)
+    
+    To be useful, this should include basis set information, electronCount, energies, 
+    """
+
+    electronCount: int = Field(..., description="Number of electrons in the species")
+    energies: Optional[List[float]] = Field(None, description="List of energies for the molecular orbitals (in eV)")
+    moCoefficients: Optional[List[float]] = Field(None, description="List of coefficients for restricted molecular orbitals, i.e., alpha=beta (requires BasisSet to be present).")
```

Comment:
How does this work w.r.t. shape? Are they flattened?

Reply:
Yes, they're flattened. At the moment, only the last MO coefficients are saved:
https://github.com/OpenChemistry/avogadro-cclib/blob/cc14cafcdfc1c27858ac1e4a64c010ca95467310/utils.py#L41

(Support for storing multiple sets of energies and moCoefficients would be welcome, e.g., to follow an IRC.)